### PR TITLE
Private routes

### PR DESCRIPTION
**Pull request by nyyshaaa**
Files to check :authentication ,private routes, profile tabs.
https://deploy-preview-3--vocal-torrone-b72432.netlify.app/

### CODE REVIEW

File: src/pages/Auth/Login.jsx [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 import "../../styles/common.css";
 import AuthStyles from  "./Auth.module.css";
 
 const Login = () => {
-    const {user,setUser}=useAuth();
+    const {user,setUser,seteToken}=useAuth();
     const navigate=useNavigate(); //useNavigate hook that returns navigate function
+    const location=useLocation();
+    // const emailRef=useRef();
+    // const passwordRef=useRef();
```
Comment (anshtripathi01) [inline src/pages/Auth/Login.jsx:15]: If this commented lines is not required remove it.

File: src/pages/Auth/Signup.jsx [hunk 2]
```diff
@@ -29,6 +30,15 @@ const Signup = () => {
...
+            if(response.status===201){
+                localStorage.setItem("user",JSON.stringify(response.data.foundUser));
+                localStorage.setItem("eToken",response.data.encodedToken);
+                seteToken(()=>response.data.encodedToken)
+                navigate("/")
+            }
+            else{
+                navigate("/signup");
+            }
             console.log(response);
```
Comment (anshtripathi01) [inline src/pages/Auth/Signup.jsx:42]: If console.log is not required here remove it

**Review by anshtripathi01 — APPROVED** [review]
Keep above changes otherwise 
folder structure code format
 and UI Looks good.

File: src/pages/Auth/Signup.jsx (hunk 2 above)
Comment (nyyshaaa) [reply src/pages/Auth/Signup.jsx:42]: okeiy will remove it,while refactor.
thanks for review!

File: src/pages/Auth/Login.jsx [hunk 3]
```diff
@@ -1,15 +1,18 @@
 import {useEffect} from 'react';
 import axios from 'axios';
-import { Link,useNavigate } from "react-router-dom";
+import { Link,useNavigate,useLocation } from "react-router-dom";
 import { Header } from '../../components/Header/Header';
 import { useAuth } from '../../context/AuthContext';
 import "../../styles/spaces.css";
 import "../../styles/common.css";
```
Comment (Sharmaryan) [inline src/pages/Auth/Login.jsx:7]: for import you can use [absolute path](https://dev.to/emanuelone/how-to-use-absolute-path-in-reactjs-503i#:~:text=Absolute%20Path%20is%20when%20you,located%20in%20the%20project%20structure.)

File: src/pages/Profile/Profile.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { useNavigate } from "react-router-dom";
+import { Header } from "../../components/Header/Header";
+import { Tabs } from "../../components/Tabs/Tabs";
+import { useAuth } from "../../context/AuthContext";
+import TabStyles from '../../components/Tabs/Tabs.module.css';
+
+
```
Comment (Sharmaryan) [inline src/pages/Profile/Profile.jsx:7]: you can do one thing, as you are importing from the components why not to make a single file and import from there.
[read here](https://dev.to/justmyrealname/organize-react-components-better-with-barrel-exports-4fno)

File: src/pages/Profile/Profile.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    return (
+        <div>
+            <Header />
+            <Tabs>
+                <div label="Profile">
+                    <p>UserEmail: {user.email}</p>
+                    <p>UserName: {user?.userName}</p>
+                </div>
+                <div label="LogOut"><button className={`${TabStyles.tabBtn}`} onClick={() => logOutHandler()}>Log Out</button></div>
```
Comment (Sharmaryan) [inline src/pages/Profile/Profile.jsx:27]: destrucuring will make your code more readable

**Review by Sharmaryan — APPROVED** [review]

File: src/pages/Auth/Login.jsx (hunk 3 above)
Comment (nyyshaaa) [reply src/pages/Auth/Login.jsx:7]: i tried last time was getting some errors after that, try again after reading about it .

File: src/pages/Profile/Profile.jsx (hunk 4 above)
Comment (nyyshaaa) [reply src/pages/Profile/Profile.jsx:7]: okeiy will do during refactor. thanks for suggestion.

File: src/pages/Profile/Profile.jsx (hunk 5 above)
Comment (nyyshaaa) [reply src/pages/Profile/Profile.jsx:27]: okeiy actually i haven't added reducer yet, so was thinking to refactor after that.
thanks for review!

**Pull request merged**